src/main/java/seedu/address/model/ModelHistoryManager.java

```diff
@@ -125,40 +129,88 @@ public ModelHistoryRecord undo() throws AlfredModelHistoryException {
             commandHistory.add(new CommandRecord((j - currentIndex),
-                    futureCommand.getClass().getSimpleName(), CommandRecord.CommandType.FUTURE));
+                                                 futureCommand.getClass().getSimpleName(),
+                                                 CommandRecord.CommandType.REDO));
         }
 
-        Command currCommand = this.history.get(currentIndex).getCommand();
-        commandHistory.add(new CommandRecord(currentIndex,
-                currCommand.getClass().getSimpleName(), CommandRecord.CommandType.CURR));
+        //Set Current Delimiter
+        commandHistory.add(CommandRecord.getCurrentStatePoint());
 
         //Obtain Undo-able Command History
         int index = 1;
-        for (int j = this.history.indexOf(this.current) - 1; j >= 0; j--) {
+        for (int j = this.history.indexOf(this.current); j >= 1; j--) {
             Command histCommand = this.history.get(j).getCommand();
-            if (histCommand == null) {
-                commandHistory.add(new CommandRecord(CommandRecord.CommandType.END));
-            } else {
-                commandHistory.add(new CommandRecord(index,
-                        histCommand.getClass().getSimpleName(),
-                        CommandRecord.CommandType.FUTURE));
-            }
+            commandHistory.add(new CommandRecord(index,
+                                                 histCommand.getClass().getSimpleName(),
+                                                 CommandRecord.CommandType.UNDO));
             index++;
         }
+        commandHistory.add(CommandRecord.getUndoEndPoint());
+        return commandHistory;
+    }
+
+    /**
+     * Returns a String representing undo-able and redo-able Command History, separated by `=` delimiter.
+     * @return String representing undo-able and redo-able Command History.
+     */
+    public String getCommandHistoryString() {
+        String commandHistory = "-------------<< Cannot Redo Beyond This Point >>-------------\n";
+        //Obtain Redo-able Command History
+        int currentIndex = this.history.indexOf(this.current);
+        for (int j = this.history.size() - 1; j > currentIndex; j--) {
+            Command futureCommand = this.history.get(j).getCommand();
+            commandHistory += ((j - currentIndex) + ": " + futureCommand.getClass().getSimpleName());
+        }
+
+        //Delimiter
+        commandHistory += "=====================<< Current State >>=====================\n";
+
+        //Obtain Undo-able Command History
+        int index = 1;
+        for (int j = this.history.indexOf(this.current); j >= 1; j--) {
+            Command histCommand = this.history.get(j).getCommand();
+            commandHistory += (index + ": " + histCommand.getClass().getSimpleName() + "\n");
+            index++;
+        }
+        commandHistory += "-------------<< Cannot Undo Beyond This Point >>-------------\n";
```

Comment:
is there a difference between ---- and =====

Reply:
aesthetics